### ISSUE TITLE
Handle errors in App::configure()

### ISSUE THREAD

**Issue by ghost** [actix/actix-web#790]
This is not an issue, more like a question, but if it's not feasible then maybe it might become an issue... 

I am wondering, is there a builtin-way in `actix_web` to handle errors when doing `App::configure()` ? In my case, I'm creating an `actix_web::fs::StaticFile` handler, the creation of which might fail. I made my `configure` method return a `Result<App<MyConfig>, MyError>` instead of `App<MyConfig>`, though I fail to see how to properly catch the error.

Please consider the following:
```
server::new(move || {
        App::with_state(inside_config.clone())
            .middleware(auth::Auth)
            .middleware(middleware::Logger::default())
            .configure(configure_app)
    })
    .bind(socket_addresses.as_slice())
    .start();
```

This works fine if `configure_app` has the following signature: 
```
fn configure_app(app: App<MyConfig>) -> App<MyConfig>
```

However, I am in this case: 
```
fn configure_app(app: App<MyConfig>) -> Result<App<MyConfig>, MyError>
```

The only way I can make this work is by doing this:
```
server::new(move || {
        App::with_state(inside_config.clone())
            .middleware(auth::Auth)
            .middleware(middleware::Logger::default())
            .configure(|app| configure_app(app).expect("oops"))
    })
    .bind(socket_addresses.as_slice())
    .start();
```

Though this will panic, while I would like to handle the error so I can display it to the user. Something like `.configure(configure_app)?` for instance, or `.configure(|app| configure_app(app)?)`.

I tried to create my app and my config before the `server::new()` call, but then Rust yells at me because `App<MyConfig>` doesn't implement `Clone`.

So, in short, is there a way to make this work ? If not, wouldn't it be interesting to be able to handle `Result` in the `configure()` method ?

**Comment by fafhrd91** [comment]
i am not sure how actix would handle errors. actix starts multiple threads and only then construct app instance in each of those threads.

**Comment by fafhrd91** [comment]
also 0.7 is in maintenance mode.
